app/models/provider_tag_mapping/mapper.rb

```diff
@@ -38,14 +42,52 @@ def initialize(mappings)
       )
     end
 
+    def case_insensitive_labels?
+      @parameters&.fetch_path(:case_insensitive_labels)
```

Comment:
Perhaps `@parameters&.[](:case_insensitive_labels)` ?

Reply:
Can `@parameters` even be nil?  If we can just ensure it's never nil in the initializer, then we don't even need the &. checks all over the place.